impeller/entity/contents/filters/matrix_filter_contents.cc

```diff
@@ -25,7 +24,35 @@ std::optional<Snapshot> MatrixFilterContents::RenderFilter(
     const Entity& entity,
     const Matrix& effect_transform,
     const Rect& coverage) const {
-  return inputs[0]->GetSnapshot(renderer, entity);
+  auto snapshot = inputs[0]->GetSnapshot(renderer, entity);
+  if (!snapshot.has_value()) {
+    return std::nullopt;
+  }
+
+  auto transform = GetTransform(entity.GetTransformation()).Invert();
+  transform = matrix_ * transform;
+  transform = GetTransform(entity.GetTransformation()) * transform;
+  snapshot->transform = transform * snapshot->transform;
```

Comment:
Here and below: Can we throw this arithmetic on one line to make the multiplication order easier to read?

```
snapshot->transform = entity.GetTransformation() * matrix_ * entity.GetTransformation().Invert() * snapshot->transform;
```

Reply:
Done.